### PR TITLE
service/kv: adjust WriteConflict error in KeyError

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi，
     This PR adjust `WriteConflict` error in KeyError, it moves from retryable to conflict field. @disksing @tiancaiamao @breeswish PTAL

### CODE REVIEW

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -1265,8 +1265,22 @@ fn extract_key_error(err: &storage::Error) -> KeyError {
...
+        storage::Error::Txn(TxnError::Mvcc(MvccError::WriteConflict {
+            start_ts,
+            conflict_ts,
+            ref key,
+            ref primary,
+        })) => {
+            let mut write_conflict = WriteConflict::new();
+            write_conflict.set_start_ts(start_ts);
+            write_conflict.set_conflict_ts(conflict_ts);
+            write_conflict.set_key(key.to_owned());
```
Comment (tiancaiamao) [inline src/server/service/kv.rs]: AFAIK, `key` is encoded while `primary` is not,  in the old error string ?
And I expect both of them to be not encoded. @AndreMouche  @disksing

**Comment by breeswish** [comment]
LGTM.  So we have `WriteConflict` for a long time but it is not used 😢 ?

File: src/server/service/kv.rs (hunk 1 above)
Comment (disksing) [reply src/server/service/kv.rs]: I have no idea. @AndreMouche please ease his concern.

**Comment by AndreMouche** [comment]
No, it's used in tidb @breeswish

**Comment by AndreMouche** [comment]
The related PR in tidb: https://github.com/pingcap/tidb/pull/6457

File: src/server/service/kv.rs (hunk 1 above)
Comment (AndreMouche) [reply src/server/service/kv.rs]: comment addressed @disksing @tiancaiamao

**Pull request merged**